src/compiler/scala/tools/nsc/classpath/FileUtils.scala

```diff
@@ -24,7 +24,7 @@ object FileUtils {
   implicit class AbstractFileOps(val file: AbstractFile) extends AnyVal {
     def isPackage: Boolean = file.isDirectory && mayBeValidPackage(file.name)
 
-    def isClass: Boolean = !file.isDirectory && (file.hasExtension("class") || file.hasExtension("sig"))
+    def isClass: Boolean = !file.isDirectory && (file.hasExtension("class") || file.hasExtension("sig") || file.hasExtension("tasty"))
```

Comment:
Until now, tasty reading worked like this

```scala
class A:
  class B
```

Scala 3 produces 3 files: `A.class`, `A$B.class` and `A.tasty`. The classpath picks up `A.class` and then the classfile parser identifies `A.class` as a Scala 3 classfile, gets the `A.tasty` file and invokes the tasty reader.

With pipelining there's only the tasty file, so the classpath needs to find those.

I worry about the non-pipelining case. The classpath will now find two files (`A.class` and `A.tasty`) for the top-level class `A`. How is this case handled?

Maybe you can remove the existing code in the classfile parser to identify Scala 3 `.class` files and teach the classpath to always pick the `A.tasty` file if it exists?

Reply:
so in the present PR I guess its non-deterministic which file is selected, (of course in the end the same TASTy file will get read), the tests in the `tasty` project cover this mixed class/tasty classpath scenario.

> Maybe you can remove the existing code in the classfile parser to identify Scala 3 .class files and teach the classpath to always pick the A.tasty file if it exists?

This should be possible - or its what Scala 3 does anyway

Reply:
> I guess its non-deterministic which file is selected

Or maybe two symbols are created? Either way, we should keep it deterministic.